x/concentrated-liquidity/swaps.go

```diff
@@ -342,9 +365,13 @@ func (k Keeper) calcOutAmtGivenIn(ctx sdk.Context,
 		}
 	}
 
+	if err := k.chargeFee(ctx, poolId, sdk.NewDecCoinFromDec(tokenInMin.Denom, swapState.feeGrowthGlobal)); err != nil {
+		return writeCtx, sdk.Coin{}, sdk.Coin{}, sdk.Int{}, sdk.Dec{}, sdk.Dec{}, err
+	}
+
 	// coin amounts require int values
 	// round amountIn up to avoid under charging
-	amt0 := tokenAmountInAfterFee.Sub(swapState.amountSpecifiedRemaining).RoundInt()
+	amt0 := tokenAmountInSpecified.Sub(swapState.amountSpecifiedRemaining).RoundInt()
```

Comment:
Just to clarify, these two are equivalent since we charge fees stepwise in `amountSpecifiedRemaining` instead of upfront in `tokenAmountInAfterFee` right?

Reply:
Yes, this should be just a name change